### DISCUSSION TITLE
Hub Offline

### DISCUSSION THREAD

**Discussion by GlennGoddard** [briis/hass-weatherflow2mqtt#178] [Ideas]
So, the good news is that if the Hub is offline for 8+ hours, as soon as it comes back then this integration immediately resumes sending data.

However, since all sensors rely on receiving some UDP input in order to output sensor, no output of any type is sent or received.  This is ok, since there are only a few senors that don't rely station/hub data.  I'm ok with not getting Solar Elevation or Solar Insolation which could be done since this is a limited occurrence event; but, should we update HUB/Station status if data not received in an hour to being offline (or create a new one, I'm leaning this way).  One hour offline is the time span that WeatherFlow uses to send a notification.  I use "secondary_info: last-changed" to see when the last update in data is.

This is the first time my hub has ever been offline (from HA).

I can't say if the weather forecast updated or not during that time since it was sunny all day.

For me, once I looked, it was obvious it was offline but it would be nice to a sensor stating status of offline/online.  Probably at an interval at least double the worst battery condition.  I'm not sure about implementation, whether it should be a countdown time that is reset on every UDP message or if this is even practical to explore.

So, rambling complete...thoughts?

**Comment by briis** [comment]
Sounds like a good idea. I was just thinking - the Hub Device has a Uptime Sensor `sensor.hub_hb_SERIAL_NO_status`. Maybe we should test what happens to this value, if the Hub is down? Just before trying to create something ourself?

**Reply by GlennGoddard** [reply]
Hub status does not change until connection is reestablished.
